modules/git/git.go

```diff
@@ -166,6 +168,47 @@ func InitSimple(ctx context.Context) error {
 
 var initOnce sync.Once
 
+func initFixGitHome117rc() error {
+	// Gitea 1.17-rc uses "setting.RepoRootPath" for Git HOME, which is incorrect.
+	// Do this check to make sure there is no legacy file in the RepoRootPath. This check might be able to be removed with 1.19 release.
+
+	// remove the auto generated git config file (it will be moved to new home)
+	gitConfigNewPath := filepath.Join(HomeDir(), ".gitconfig")
+	gitConfigLegacyPath := filepath.Join(setting.RepoRootPath, ".gitconfig")
+	if ok, err := util.IsExist(gitConfigLegacyPath); ok && err == nil {
+		if err = os.MkdirAll(HomeDir(), os.ModePerm); err != nil {
+			return err
+		}
+		if ok, err = util.IsExist(gitConfigNewPath); !ok && err == nil {
+			err = util.CopyFile(gitConfigLegacyPath, gitConfigNewPath)
+		} else {
+			err = util.CopyFile(gitConfigLegacyPath, gitConfigNewPath+".bak")
+		}
+		if err != nil {
+			return err
+		}
+		_ = os.Remove(gitConfigLegacyPath)
```

Comment:
I really don't think we should be removing anything or doing anything to these 1.17rc-1 and dev placed directories.

The whole problem with that scheme was that we were not completely certain that these were the correct place. 

The best we should do is warn that there was this data there and/or create a doctor command for migrating things. The warning should only be there until 1.17 is actually released too.

Reply:
Sorry but I do not understand why the removing is incorrect, what damage would it cause. IMO the removing is carefully designed, it won't break any user's file.

Or could you edit on this PR directly for what you want?

Reply:
how do you know that any of these are the user's files?

Reply:
Please provide a sample to show how they are affected or damaged then make the user hurt.

Reply:
Honestly I just think we should leave them where they are and at most just add a warning to the user - (up until 1.17 is released) that says that we found files there that may relate to old config but we can't be sure and get the user to make that determination themselves.

Reply:
TBH I have no confidence to make a commit satisfying your idea because I am not sure in your opinion what should be done while what shouldn't, you can edit on the PR directly.

Reply:
Let me be clearer:

We have changed this GIT_HOME because we have NO confidence that setting this to REPO_ROOT_PATH was correct.

We cannot therefore be certain that anything that we find in REPO_ROOT_PATH is correct.

Your code is:

* copying stuff from REPO_ROOT_PATH arbitrarily
* then deleting stuff from REPO_ROOT_PATH arbitrarily

It is then deleting `.ssh` and `.gnupg` if they're empty directories and fatalling asserting that they're incorrect if they're non-empty. `.ssh` and `.gnupg` are perfectly valid usernames - and I see no reason to make them invalid.

This check function should at best be a doctor command but it certainly should not be running by default.

Reply:
These code is removed. But I would suggest you to check the `RepoRootPath/.gnupg/pubring.kbx` to see whether the directory has been used as a gnupg home (for 1.17rc1 users), to avoid potential key leaks in the future.